dist/dialog.linker.build.js

```diff
@@ -198,14 +216,30 @@ window.jQuery = window.jQuery || window.shoestring;
 			}
 		});
 
+	// if the hash matches an ajaxlink's url, open it by triggering a click on the ajaxlink
+	$( w ).bind( "hashchange load", function(){
+		var hash = w.location.hash.split( "#" ).pop();
+		var id = hash.replace( /-dialog$/, "" );
+		var $ajaxLink = $( 'a[href="' + id +'"][data-dialog-link]' );
+		// if the link specified nohistory, don't click it
+		var nohistory = $ajaxLink.is( "[data-dialog-nohistory]" );
+		var $dialogInPage = $( '.dialog[id="' + id + '"]' );
+		if( $ajaxLink.length && !nohistory && !$dialogInPage.length ){
+			$ajaxLink.eq( 0 ).trigger( "click" );
+		}
```

Comment:
So John this ONLY covers the case when the hash matches an ajax dialog link that has not yet been clicked, and thus, has no dialog to open yet. It'll only happen once because once the dialog is created, that `$dialogInPage` var will be true and prevent this behavior from applying the click. 

After it's created, it'll still be opened by our ordinary dialog hashchange listener.